src/hotspot/cpu/x86/methodHandles_x86.cpp

```diff
@@ -122,17 +122,63 @@ void MethodHandles::verify_ref_kind(MacroAssembler* _masm, int ref_kind, Registe
   __ bind(L);
 }
 
-#endif //ASSERT
+void MethodHandles::verify_method(MacroAssembler* _masm, Register method, Register temp, vmIntrinsics::ID iid) {
+  BLOCK_COMMENT("verify_method {");
+  __ verify_method_ptr(method);
+  if (VerifyMethodHandles) {
+    Label L_ok;
+
+    const Register method_holder = temp;
```

Comment:
`assert_different_registers` missing. Or you don't need it?

Reply:
Added.